rotkehlchen/data_import/importers/kucoin.py

```diff
@@ -0,0 +1,111 @@
+    ) -> Trade:
+        """
+        Consume entries of kucoin order history csv
+
+        May raise:
+        - KeyError
+        - DeserializationError
+        - UnknownAsset
+        - ValueError
+        """
+        if file_type == KucoinFileType.TRADES:
+            date = csv_row['tradeCreatedAt']
+            base, quote = csv_row['symbol'].split('-')
+            trade_type_raw = csv_row['side']
+            amount = csv_row['size']
+            rate = csv_row['price']
+            fee = get_key_if_has_val(csv_row, 'fee')
+            fee_currency = get_key_if_has_val(csv_row, 'feeCurrency')
+        else:
+            date = csv_row['Time']
+            base, quote = csv_row['Pair'].split('/')
+            trade_type_raw = csv_row['Side']
+            amount = csv_row['Amount']
+            rate = csv_row['FilledPrice']
+            fee = get_key_if_has_val(csv_row, 'Fee')
+            fee_currency = get_key_if_has_val(csv_row, 'FeeCoin')
+        trade_type = TradeType.BUY if trade_type_raw.lower() == 'buy' else TradeType.SELL
+
+        return Trade(
+            timestamp=deserialize_timestamp_from_date(
+                date=date,
+                formatstr=timestamp_format,
+                location='Kucoin order history import',
+            ),
+            location=Location.KUCOIN,
+            base_asset=asset_from_kucoin(base),
+            quote_asset=asset_from_kucoin(quote),
+            trade_type=trade_type,
+            amount=deserialize_asset_amount(amount),
+            rate=deserialize_price(rate),
+            fee=deserialize_fee(fee) if fee else None,
+            fee_currency=asset_from_kucoin(fee_currency) if fee_currency else None,
+        )
+
+    def _import_csv(self, write_cursor: DBCursor, filepath: Path, **kwargs: Any) -> None:
+        """
+        Import trades from Kucoin. Find out which file
+        we are parsing depending on its format.
+
+        May raise:
+        - UnsupportedCSVEntry if operation not supported
+        - InputError if a column we need is missing
+        """
+        with open(filepath, encoding='utf-8-sig') as csvfile:
+            start_pos = csvfile.tell()
+            first_line = csvfile.readline()
+            if 'tradeCreatedAt,symbol,side,price,size,funds,fee,feeCurrency' in first_line:
+                file_type = KucoinFileType.TRADES
+            elif 'Time,Pair,Side,FilledPrice,FilledPriceCoin,Amount,AmountCoin,Volume,VolumeCoin,Fee,FeeCoin' in first_line:  # noqa: E501
+                file_type = KucoinFileType.TRADES_OLD
+            else:
+                raise InputError('The given kucoin CSV file format can not be recognized')
+
+            csvfile.seek(start_pos)  # go back to start of the file
+            data = csv.DictReader(csvfile)
+            for row in data:
+                try:
+                    event = self._consume_trades(row, file_type, **kwargs)
+                    self.add_trade(write_cursor, event)  # checked by if above
```

Comment:
I don't knwo what this comment means

```suggestion
                    self.add_trade(write_cursor, event)
```